### ISSUE TITLE
edd_payment_icons_callback uses value, not key to fire hook

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/easy-digital-downloads#5827]
Right now `edd_payment_icons_callback()` uses the `$option` instead of the `$key` to fire a wildcard hook:

```
foreach( $args['options'] as $key => $option ) {
     {...}
    $card = strtolower( str_replace( ' ', '', $option ) );

    if( has_filter( 'edd_accepted_payment_' . $card . '_image' ) ) {  
        $image = apply_filters( 'edd_accepted_payment_' . $card . '_image', '' );
        {...}
    }
}
```
The problem here arrises that you might setup a gateway with the following format:
`$args['options']['my-awesome-gateway'] = '支付宝'`

This would then try and fire filter:
`edd_accepted_payment_支付宝_image`
as opposed to:
`edd_accepted_payment_my-awesome-gateway_image`

This will have to be completely backwards compatible.

**Comment by cklosowski** [comment]
For reference the original PR: #4527

**Comment by cklosowski** [comment]
All existing work is in `issue/5827`

**Comment by JJJ** [comment]
I deleted the old `issue/5827` branch and replaced it with `issue/5827-2`.

I don't think we should allow filtering using both approaches, as we run the risk of one overloading the other. Instead, I propose we just take the same approach we took originally, but with `$key` instead, and also leave the old way/code intact.

This way, if anyone is using the old way, it continues to work per normal. If people use the new way, it will work, too. If someone uses both, the old way wins because it's the weirder of the 2, and back-compat is important here.
